packages/apps/explorer/src/components/search/search.tsx

```diff
@@ -0,0 +1,237 @@
+import { truncateValues } from '@/services/format';
+import { MonoSearch } from '@kadena/react-icons/system';
+import { Badge, Box } from '@kadena/react-ui';
+import { atoms } from '@kadena/react-ui/styles';
+import React, { useState } from 'react';
+
+export type SearchItemTitle =
+  | 'Account'
+  | 'Request Key'
+  | 'Block Height'
+  | 'Block Hash'
+  | 'Events';
+
+export interface ISearchItem {
+  title: SearchItemTitle;
+  disabled?: boolean;
+}
+interface ISearchComponentProps {
+  placeholder: string;
+  searchItems: ISearchItem[];
+}
+
+const SearchCombobox: React.FC<ISearchComponentProps> = ({
+  placeholder,
+  searchItems,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [searchOption, setSearchOption] = useState<number | null>(null);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [optionClicked, setOptionClicked] = useState(false);
+  const [escapePressed, setEscapePressed] = useState(false);
+
+  const setOptionsDisabledExcept = (exceptIndex: number): void => {
+    searchItems.forEach((item, index) => {
+      if (index !== exceptIndex) {
+        item.disabled = true;
+      }
+    });
+  };
+
+  const inferOption = (value: string): SearchItemTitle | undefined => {
+    if (
+      value.toLocaleLowerCase().startsWith('k:') ||
+      value.toLocaleLowerCase().startsWith('w:')
+    ) {
+      return 'Account';
+    } else if (value.includes('.')) {
+      return 'Events';
+    } else if (value.length === 43) {
+      return 'Request Key';
+    } else if (/^\d+$/.test(value)) {
+      return 'Block Height';
+    }
+
+    return undefined;
+  };
+
+  const enableAllOptions = (): void => {
+    searchItems.forEach((item) => {
+      item.disabled = false;
+    });
+  };
+
+  const handleSearch = (value: string, option: number | null): void => {};
```

Comment:
Do we need this?

Reply:
This is the function to be implemented once we have the details page and we can actually search. So I would say at the moment is not needed but it will need to be implemented in the future. Should I remove it?